libs/community/langchain_community/llms/solar.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Dict, List, Optional
+
+import requests
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.language_models import LLM
+from langchain_core.pydantic_v1 import BaseModel, Field, SecretStr, root_validator
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
+
+from langchain_community.llms.utils import enforce_stop_tokens
+
+SOLAR_SERVICE_URL_BASE = "https://api.upstage.ai/v1/solar"
+
+
+class _SolarClient(BaseModel):
+    """An API client that talks to the Solar server."""
+
+    api_key: SecretStr
+    """The API key to use for authentication."""
+    base_url: str = SOLAR_SERVICE_URL_BASE
+
+    def completion(self, request: Any) -> Any:
+        headers = {"Authorization": f"Bearer {self.api_key.get_secret_value()}"}
+        response = requests.post(
+            f"{self.base_url}/chat/completions",
+            headers=headers,
+            json=request,
+        )
+        if not response.ok:
+            raise ValueError(f"HTTP {response.status_code} error: {response.text}")
+        return response.json()["choices"][0]["message"]["content"]
+
+
+class SolarCommon(BaseModel):
+    _client: _SolarClient
+    base_url: str = SOLAR_SERVICE_URL_BASE
+    solar_api_key: Optional[SecretStr] = Field(default=None, alias="api_key")
+    """Solar API key. Get it here: https://console.upstage.ai/services/solar"""
+    model_name: str = Field(default="solar-1-mini-chat", alias="model")
+    """Model name. Available models listed here: https://console.upstage.ai/services/solar"""
+    max_tokens = 1024
+    temperature = 0.3
+
+    class Config:
+        allow_population_by_field_name = True
```

Comment:
what is this for? any comments?

Reply:
It's a config for pydantic object. The detailed explanation lies here. https://docs.pydantic.dev/1.10/usage/model_config. In this pr, we just follow the common sense in this repo such as openai implementation.
https://github.com/langchain-ai/langchain/blob/a4a6978224f7addcb7e255c8e6e7fe517dd9955f/libs/community/langchain_community/llms/openai.py#L260